src/editors/containers/TextEditor/components/ImageSettingsModal/index.jsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Button,
+  Form,
+  Image,
+} from '@edx/paragon';
+
+import BaseModal from '../BaseModal';
+import * as module from '.';
+
+export const hooks = {
+  dimensions: () => {
```

Comment:
We'll probaly want to have the initialization of these hooks take input, for the case in which the image settings are being edited.